managedtenants/core/addons_loader/addon.py

```diff
@@ -111,6 +118,17 @@ def load_metadata(self, environment):
         return metadata
 
     def load_imageset(self, imageset_version):
+
+        # Do not allow pin-pointing a version
+        # Temporary hot-fix for https://issues.redhat.com/browse/SDA-4918
+        # See: https://issues.redhat.com/browse/MTSRE-453
+        if self.imageset_latest_only and imageset_version != "latest":
+            raise AddonLoadError(
+                "Pointing to a specific version is not allowed. Please use"
+                " only latest. See https://issues.redhat.com/browse/MTSRE-453"
+                " for more info"
+            )
+
```

Comment:
can we move this under the constructor itself i.e. under the line 
```py
...
if self.imageset_version is not None:
  if self.imageset_latest_only and imageset_version != "latest":
    raise AddonLoadError(
                "Pointing to a specific version is not allowed. Please use"
                " only latest. See https://issues.redhat.com/browse/MTSRE-453"
                " for more info"
            )
  self.imagesets_path = self.path / f"addonimagesets/{environment}"
  self.imageset = self.load_imageset(self.imageset_version)
  self.package = None
  self.bundles = None
  self.catalog_image = Image(self.imageset["indexImage"])
```

Semantically, it feels that leaving `load_imageset` to just "load imagesets" makes more sense and validation like this one should be a part of constructor (or maybe a separate method which gets called in the constructor) considering the fact that it relies upon setting up the Addon object itself more than the load_imageset step.

But this is my POV, not a hard requirement. So feel free to resolve this suggestion straight away if you think that  the original implementation makes more sense. I am happy with anything.

Rest looks good :)